<a href="https://colab.research.google.com/github/WahyuriN/UTS_datasience/blob/main/pertemuan2_Wahyu_Ari_Nugroho_240401010173.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Nama : Wahyu Ari Nugroho
# NIM : 240401010173
# Kelas : IF401

================================================

PERTEMUAN 2: Struktur Data Python, NumPy dan Pandas

================================================

# Import Library

In [12]:
import pandas as pd
import numpy as np


# Muat Dataset Titanic

In [13]:

url = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(url)

# Tampilkan head() dan info()

In [14]:
print(df.head())

   PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  


In [15]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB
None


# Tampilkan jumlah baris dan kolom, dan tampilkan nama kolom

In [16]:
print(df.shape, df.columns.tolist())

(891, 12) ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']


# Tampilkan missing values

In [17]:
print(df.isnull().sum())

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64


# Tampilkan persentase penumpang selamat

In [18]:
print(df['Survived'].mean())

0.3838383838383838


# Filter penumpang wanita kelas 1, jumlah yang selamat, dan Rata-rata usia

In [19]:
wk1 = df[(df['Sex']=='female') &
(df['Pclass']==1)]
print(wk1['Survived'].mean(), wk1['Age'].mean())



0.9680851063829787 34.61176470588235


# Menjalankan groupby('Pclass')['Survived'].mean()

In [20]:
print(df.groupby('Pclass')['Survived'].mean())


Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64


#Jawaban


1.    jumlah baris dan kolom dataset Titanic adalah 891  baris dan 12 kolom
2.   kolom yang memiliki nilai kosong (missing value) adalah kolom Age, Cabin, dan Embarked
3. penumpang yang selamat kurang lebih 38%
4. Rata-rata usia penumpang wanita kelas 1 yang
selamat adalah usia 34 tahun
5. Kelas penumpang yang memiliki tingkat keselamatan tertinggi adalah kelas 1 dengan persentase sekitar 0,63 atau 63%

